### PR TITLE
Invitationemaildowncase

### PR DESCRIPTION
**Pull request by mahtai**
Invitation emails need to be all lowercase to match emails in the user database.

### CODE REVIEW

File: app/controllers/invitations_controller.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ def create
     @invitation.inviter = current_user
     @invitation.token = Digest::SHA1.hexdigest([@invitation.sender_id, Time.now, rand].join)
     @team = Team.find(@invitation.sender_id)
+    @invitation.email = @invitation.email.downcase!
```
Comment (mzagaja) [inline app/controllers/invitations_controller.rb]: Consider using a before_filter in the model to handle this situation: http://guides.rubyonrails.org/action_controller_overview.html#filters

**Comment by mzagaja** [comment]
Also you can probably refactor https://github.com/codeforboston/cambridge_energy_app/blob/master/app/controllers/invitations_controller.rb#L29-L31 into the model as well using similar techniques.

**Comment by mzagaja** [comment]
Actually after looking into this further it probably makes sense to do these things in user_mailer.rb

File: app/controllers/invitations_controller.rb [hunk 2]
```diff
@@ -26,24 +27,21 @@ def edit
...
+    @receiver = User.find_by_email(@invitation.email)
+    if @receiver
+      @invitation.receiver_id = @receiver.id
     end
     respond_to do |format|
       if @invitation.save
-        if receiver
-          UserMailer.existing_user_invite_email(@invitation).deliver_now
+        if @receiver
+          #UserMailer.existing_user_invite_email(@invitation).deliver_now
```
Comment (mzagaja) [inline app/controllers/invitations_controller.rb]: Commenting this out prevented the invitation email from firing.

File: app/controllers/invitations_controller.rb [hunk 3]
```diff
@@ -26,24 +27,21 @@ def edit
...
           message = 'Invitation sent to existing user'
         else
-          UserMailer.new_user_invite_email(@invitation, new_user_registration_path(:invitation_token => @invitation.token)).deliver_now
+          #UserMailer.new_user_invite_email(@invitation).deliver_now
```
Comment (mzagaja) [inline app/controllers/invitations_controller.rb]: Commenting this out prevented the invitation email from firing

**Comment by mzagaja** [comment]
When I earlier linked to the before_filter what I was really hoping to link to was ActiveRecord Callbacks: http://guides.rubyonrails.org/active_record_callbacks.html as some of the stuff in the controller can probably be refactored into the model using those. http://www.theodinproject.com/ruby-on-rails/active-record-callbacks might also be helpful on this. It is not required/urgent but we should do that at some point.

**Comment by mzagaja** [comment]
This breaks two tests:
- [ ] rspec ./spec/controllers/invitations_controller_spec.rb:46 # InvitationsController POST #create creates a new invitation
- [ ] rspec ./spec/controllers/invitations_controller_spec.rb:70 # InvitationsController DELETE #destroy reciever declines invitation is successful

File: app/views/teams/inviting.html.erb [hunk 4]
```diff
@@ -1,6 +1,11 @@
+<p id="notice"><%= notice %></p>
+
 <p>
-    <strong>Users invited to join Team <%= @team.name %></strong>
+    <strong>Users invited to join <%= @team.name %></strong>
 </p>
+<% if @invitations.nil? || @invitations.empty? %>
```
Comment (mpautasso) [inline app/views/teams/inviting.html.erb:6]: I think it's enough to check `@invitations.empty?` because `invitations` is an array.

**Comment by mzagaja** [comment]
Closing because it is rendered moot by the use of devise_invitable I merged in.

**Pull request closed without merging**